Denis_Protsenko.py

```diff
@@ -0,0 +1,218 @@
+from pydantic import BaseModel
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from typing import Union
+
+
+# Абстрактний клас
+class PublicModel(ABC):
+    @abstractmethod
+    def book_info(self):
+        pass
+
+
+# Клас моделі книги
+class BookModel(BaseModel):
+    title: str
+    author: str
+    year: int
+
+
+# Клас моделі журналу
+class MagazineModel(BaseModel):
+    title: str
+    author: str
+    year: int
+
+
+# Клас книги
+class Book(PublicModel):
+    def __init__(self, book_model: BookModel):
+        self.title = book_model.title
+        self.author = book_model.author
+        self.year = book_model.year
+
+    def book_info(self):  # Implementing the abstract method
+        return f"{self.title}, Автор: {self.author}, Рік видання: {self.year}"
+
+    def __eq__(self, other):
```

Comment:
Good idea to make class as hashable to use as dict key.
But I haven't found where you used this oportunity.
If you want, we can discuss this solution more.